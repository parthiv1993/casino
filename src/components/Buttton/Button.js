import PropTypes from 'prop-types';
import StyledButton from './Button.style';

const Button = ({ children, fullWidth, buttonType, onClick, ...restProps }) => {
  return (
    <StyledButton fullWidth={fullWidth} buttonType={buttonType} onClick={onClick} {...restProps}>
      {children}
    </StyledButton>
  );
};

Button.propTypes = {
  children: PropTypes.node,
  fullWidth: PropTypes.bool,
  onClick: PropTypes.func,
  buttonType: PropTypes.oneOf(['primary', 'secondary'])
};

Button.defaultProps = {
  children: null,
  buttonType: 'primary',
  fullWidth: false,
  onClick: () => {}
};

export default Button;
