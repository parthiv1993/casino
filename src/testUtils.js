import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import light from './contexts/theme/light';

export const renderComponentWithLightTheme = (children) => {
  return render(<ThemeProvider theme={light}>{children}</ThemeProvider>);
};

export const lowerCaseAllKeys = (props = {}) => {
  const lowercaseObject = {};
  Object.keys(props).map((key) => (lowercaseObject[key.toLowerCase()] = props[key]));

  return lowercaseObject;
};

export const MockIcon = (props) => <div component="Icon" {...lowerCaseAllKeys(props)} />;
