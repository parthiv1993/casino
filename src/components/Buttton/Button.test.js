import { fireEvent } from '@testing-library/react';
import { renderComponentWithLightTheme } from '../../testUtils';
import Button from './Button';

describe('Button', () => {
  it('should match snapshot', () => {
    const { container } = renderComponentWithLightTheme(<Button>Button Text</Button>);
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot when it is fullwidth', () => {
    const { container } = renderComponentWithLightTheme(<Button fullWidth>Button Text</Button>);
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot when it is secondaryButton', () => {
    const { container } = renderComponentWithLightTheme(
      <Button buttonType="secondary">Button Text</Button>
    );
    expect(container).toMatchSnapshot();
  });

  it('should call onClick when button is clicked', () => {
    const mockOnClick = jest.fn();
    const { getByTestId } = renderComponentWithLightTheme(
      <Button data-testid="test-button" onClick={mockOnClick}>
        Button Text
      </Button>
    );
    fireEvent.click(getByTestId('test-button'));
    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });
});
