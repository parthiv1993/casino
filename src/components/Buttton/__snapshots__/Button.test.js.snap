// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
.c0 {
  background: white;
  border-width: 1px;
  border-color: rgba(0,0,0,0.15);
  color: #444;
  cursor: pointer;
  padding: 0.5em;
  border-radius: 4px;
}

.c0.c0:focus {
  outline: 2px solid blue;
}

<div>
  <button
    class="c0"
  >
    Button Text
  </button>
</div>
`;

exports[`Button should match snapshot when it is fullwidth 1`] = `
.c0 {
  width: 100%;
  background: white;
  border-width: 1px;
  border-color: rgba(0,0,0,0.15);
  color: #444;
  cursor: pointer;
  padding: 0.5em;
  border-radius: 4px;
}

.c0.c0:focus {
  outline: 2px solid blue;
}

<div>
  <button
    class="c0"
  >
    Button Text
  </button>
</div>
`;

exports[`Button should match snapshot when it is secondaryButton 1`] = `
.c0 {
  background: #1b1c1d;
  border-width: 0px;
  border-color: none;
  color: white;
  cursor: pointer;
  padding: 0.5em;
  border-radius: 4px;
}

.c0.c0:focus {
  outline: 2px solid blue;
}

<div>
  <button
    class="c0"
  >
    Button Text
  </button>
</div>
`;
